### PR TITLE
fix(http/github): Preserve search params for paginated requests

### LINKED ISSUE #13474: Github paginationField flag stopped working
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Describe the bug

The paginationField option used when fetching the list of available repositories from Github does not work after the refactor in PR #13416 was merged. 

Snippets: 
Previous code: https://github.com/renovatebot/renovate/blob/e3d213e7c5f4451100f081737d9fbf2f58f5da26/lib/util/http/github.ts#L236-L249

New code: https://github.com/renovatebot/renovate/blob/030b1a61a45a98557ddfe22732ffa1807619cba5/lib/util/http/github.ts#L239-L258

The constructed array is right, but the result is ignored. 

### Relevant debug logs

_No response_

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

`delete url.search` did nothing, but after refactoring to `url.search = ''` it started actually do the thing. This leads to errors, do we don't touch any url parameters other than `page`.

## Context:

- Closes #13474

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/github.ts [hunk 1]
```diff
@@ -210,31 +210,26 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
...
         // Check if result is paginated
         const pageLimit = opts.pageLimit ?? 10;
-        const linkHeader =
-          result?.headers?.link &&
-          parseLinkHeader(result.headers.link as string);
+        const linkHeader = is.string(result?.headers?.link)
+          ? parseLinkHeader(result.headers.link)
+          : null;
         if (linkHeader?.next && linkHeader?.last) {
           let lastPage = +linkHeader.last.page;
```
Comment (JamieMagee) [inline lib/util/http/github.ts:217]: I know it's not related to your changes, but it looked funky 😅 

```suggestion
          let lastPage = Number.parseInt(linkHeader.last.page);
```

**Review by JamieMagee — DISMISSED** [review]

File: lib/util/http/github.ts [hunk 2]
```diff
@@ -210,31 +210,26 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
       if (opts.paginate) {
         // Check if result is paginated
         const pageLimit = opts.pageLimit ?? 10;
-        const linkHeader =
-          result?.headers?.link &&
-          parseLinkHeader(result.headers.link as string);
+        const linkHeader = is.string(result?.headers?.link)
+          ? parseLinkHeader(result.headers.link)
+          : null;
```
Comment (viceice) [inline lib/util/http/github.ts:215]:
```suggestion
        const linkHeader = parseLinkHeader(result?.headers?.link);
```
`parseLinkHeader` will handle undefined or null

File: lib/util/http/github.ts [hunk 3]
```diff
@@ -210,31 +210,26 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
...
         if (linkHeader?.next && linkHeader?.last) {
-          let lastPage = +linkHeader.last.page;
+          let lastPage = Number(linkHeader.last.page);
```
Comment (viceice) [inline lib/util/http/github.ts:217]:
```suggestion
          let lastPage = parseInt(linkHeader.last.page, 10);
```
 I think this is more safe.

File: lib/util/http/github.ts [hunk 4]
```diff
@@ -210,31 +210,26 @@ export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
...
           if (!process.env.RENOVATE_PAGINATE_ALL && opts.paginate !== 'all') {
             lastPage = Math.min(pageLimit, lastPage);
           }
-          const pageNumbers = Array.from(
-            new Array(lastPage),
-            (x, i) => i + 1
-          ).slice(1);
-          const queue = pageNumbers.map(
-            (page) => (): Promise<HttpResponse<T>> => {
+          const queue = Array.from(new Array(lastPage), (_, i) => `${i + 1}`)
```
Comment (viceice) [inline lib/util/http/github.ts:222]: That looks really ugly, we should replace it with a generator function like in test in a future PR.

**Review by viceice — COMMENTED** [review]
Some small things, otherwise LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**